src/persist-client/src/internal/trace.rs

```diff
@@ -2185,4 +2486,38 @@ pub(crate) mod tests {
             vec![req015, req(0, 1)]
         );
     }
+
+    #[mz_ore::test]
+    fn construct_batch_with_runs_replaced_test() {
+        proptest!(|(
+            (batch, to_replace, range) in (any_hollow_batch::<u64>(), any_hollow_batch_with_exact_runs::<u64>(1))
+            .prop_filter("non-empty batch", |(batch, _)| batch.run_meta.len() >= 1)
+            .prop_flat_map(|(batch, to_replace)| {
+                let batch_len = batch.run_meta.len();
+                (0..batch_len).prop_flat_map(move |start| {
+                let max_len = batch_len - start;
+                let batch_clone = batch.clone();
+                let to_replace_clone = to_replace.clone();
+                (1..=max_len).prop_map(move |len| {
+                    let range = start..(start + len);
+                    (batch_clone.clone(), to_replace_clone.clone(), range)
+                })
```

Comment:
This test still bakes in the "must replace a contiguous range of runs from the original batch" idea that the function under test no longer requires. Might as well choose a random subset instead and get better coverage?